File: backend/src/WMS.Domain/Entities/Product.cs
using WMS.Domain.Enums;

namespace WMS.Domain.Entities
{
    /// <summary>
    /// Entidade que representa um SKU (Stock Keeping Unit) / Produto no sistema
    /// Contém informações sobre produtos que podem ser armazenados nos armazéns
    /// </summary>
    public class Product : BaseEntity
    {
        /// <summary>
        /// Código único do produto (SKU)
        /// Exemplo: SKU-001-ABC
        /// </summary>
        public string Sku { get; set; } = string.Empty;

        /// <summary>
        /// Nome do produto
        /// </summary>
        public string Name { get; set; } = string.Empty;

        /// <summary>
        /// Descrição detalhada do produto
        /// </summary>
        public string? Description { get; set; }

        /// <summary>
        /// Categoria do produto (Seco, Refrigerado, Congelado, Perecível, etc)
        /// </summary>
        public ProductCategory Category { get; set; }

        /// <summary>
        /// Tipo do produto (Commodity, Fracionável, Frágil, etc)
        /// </summary>
        public ProductType Type { get; set; }

        /// <summary>
        /// Peso unitário do produto em kg
        /// </summary>
        public decimal UnitWeight { get; set; }

        /// <summary>
        /// Volume unitário do produto em m³
        /// </summary>
        public decimal UnitVolume { get; set; }

        /// <summary>
        /// Zona de armazenamento padrão para este produto
        /// (Picking, Reserva, Cross-Dock, Quarentena)
        /// </summary>
        public StorageZone DefaultStorageZone { get; set; }

        /// <summary>
        /// Indica se o produto requer rastreamento de lote
        /// </summary>
        public bool RequiresLotTracking { get; set; }

        /// <summary>
        /// Indica se o produto requer rastreamento de número de série
        /// </summary>
        public bool RequiresSerialNumber { get; set; }

        /// <summary>
        /// Período de vida útil do produto em dias (shelf life)
        /// Usado para produtos perecíveis
        /// </summary>
        public int? ShelfLifeDays { get; set; }

        /// <summary>
        /// Indica se o produto está ativo no sistema
        /// </summary>
        public bool IsActive { get; set; } = true;

        /// <summary>
        /// ID do tenant proprietário deste produto
        /// </summary>
        public int TenantId { get; set; }

        /// <summary>
        /// Temperatura mínima recomendada para armazenamento (em Celsius)
        /// Usado para produtos refrigerados/congelados
        /// </summary>
        public decimal? MinStorageTemperature { get; set; }

        /// <summary>
        /// Temperatura máxima recomendada para armazenamento (em Celsius)
        /// Usado para produtos refrigerados/congelados
        /// </summary>
        public decimal? MaxStorageTemperature { get; set; }

        /// <summary>
        /// Umidade relativa mínima recomendada (em percentual)
        /// </summary>
        public decimal? MinStorageHumidity { get; set; }

        /// <summary>
        /// Umidade relativa máxima recomendada (em percentual)
        /// </summary>
        public decimal? MaxStorageHumidity { get; set; }

        /// <summary>
        /// Indica se o produto é inflamável
        /// </summary>
        public bool IsFlammable { get; set; }

        /// <summary>
        /// Indica se o produto é tóxico ou perigoso
        /// </summary>
        public bool IsDangerous { get; set; }

        /// <summary>
        /// Indica se o produto é um medicamento (requer rastreabilidade especial)
        /// </summary>
        public bool IsPharmaceutical { get; set; }

        /// <summary>
        /// Classificação ABC do produto (baseada em giro/valor)
        /// A = Alto giro, B = Médio giro, C = Baixo giro
        /// </summary>
        public ABCClassification? ABCClassification { get; set; }

        /// <summary>
        /// Custo unitário do produto (em reais)
        /// </summary>
        public decimal? UnitCost { get; set; }

        /// <summary>
        /// Preço de venda unitário (em reais)
        /// </summary>
        public decimal? UnitPrice { get; set; }
    }
}

File: frontend/src/types/product.ts
// ============================================================================
// Tipos e Interfaces para Produtos
// ============================================================================

export enum ProductCategory {
  Dry = 1,
  Refrigerated = 2,
  Frozen = 3,
  Perishable = 4,
  Controlled = 5,
  BulkVolume = 6,
  SmallVolume = 7,
  HighValue = 8,
}

export enum ProductType {
  Commodity = 1,
  Fractionable = 2,
  Fragile = 3,
  Heavy = 4,
  Liquid = 5,
  Gaseous = 6,
  Bulky = 7,
  IsolationRequired = 8,
}

export enum StorageZone {
  Picking = 1,
  Reserve = 2,
  CrossDock = 3,
  Quarantine = 4,
  Damage = 5,
  Refrigerated = 6,
  Frozen = 7,
  Controlled = 8,
  Consolidation = 9,
  Packing = 10,
}

export enum ABCClassification {
  A = 1,
  B = 2,
  C = 3,
}

export interface Product {
  id: string;
  sku: string;
  name: string;
  category: ProductCategory;
  type: ProductType;
  unitWeight: number;
  unitVolume: number;
  defaultStorageZone: StorageZone;
  requiresLotTracking: boolean;
  requiresSerialNumber: boolean;
  shelfLifeDays: number | null;
  minStorageTemperature: number | null;
  maxStorageTemperature: number | null;
  minStorageHumidity: number | null;
  maxStorageHumidity: number | null;
  isFlammable: boolean;
  isDangerous: boolean;
  isPharmaceutical: boolean;
  abcClassification: ABCClassification | null;
  unitCost: number | null;
  unitPrice: number | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProductRequest {
  sku: string;
  name: string;
  category: number;
  type: number;
  unitWeight: number;
  unitVolume: number;
  defaultStorageZone: number;
  requiresLotTracking: boolean;
  requiresSerialNumber: boolean;
  shelfLifeDays?: number;
  minStorageTemperature?: number;
  maxStorageTemperature?: number;
  minStorageHumidity?: number;
  maxStorageHumidity?: number;
  isFlammable: boolean;
  isDangerous: boolean;
  isPharmaceutical: boolean;
  abcClassification?: number;
  unitCost?: number;
  unitPrice?: number;
}

export interface UpdateProductRequest {
  name?: string;
  category?: number;
  type?: number;
  unitWeight?: number;
  unitVolume?: number;
  defaultStorageZone?: number;
  requiresLotTracking?: boolean;
  requiresSerialNumber?: boolean;
  shelfLifeDays?: number;
  minStorageTemperature?: number;
  maxStorageTemperature?: number;
  minStorageHumidity?: number;
  maxStorageHumidity?: number;
  isFlammable?: boolean;
  isDangerous?: boolean;
  isPharmaceutical?: boolean;
  abcClassification?: number;
  unitCost?: number;
  unitPrice?: number;
  isActive?: boolean;
}

// Labels para exibição
export const ProductCategoryLabels: Record<ProductCategory, string> = {
  [ProductCategory.Dry]: 'Seco',
  [ProductCategory.Refrigerated]: 'Refrigerado',
  [ProductCategory.Frozen]: 'Congelado',
  [ProductCategory.Perishable]: 'Perecível',
  [ProductCategory.Controlled]: 'Controlado',
  [ProductCategory.BulkVolume]: 'Grande Volume',
  [ProductCategory.SmallVolume]: 'Pequeno Volume',
  [ProductCategory.HighValue]: 'Alto Valor',
};

export const ProductTypeLabels: Record<ProductType, string> = {
  [ProductType.Commodity]: 'Commodity',
  [ProductType.Fractionable]: 'Fracionável',
  [ProductType.Fragile]: 'Frágil',
  [ProductType.Heavy]: 'Pesado',
  [ProductType.Liquid]: 'Líquido',
  [ProductType.Gaseous]: 'Gasoso',
  [ProductType.Bulky]: 'Volumoso',
  [ProductType.IsolationRequired]: 'Isolamento Requerido',
};

export const StorageZoneLabels: Record<StorageZone, string> = {
  [StorageZone.Picking]: 'Picking',
  [StorageZone.Reserve]: 'Reserva',
  [StorageZone.CrossDock]: 'Cross-Dock',
  [StorageZone.Quarantine]: 'Quarentena',
  [StorageZone.Damage]: 'Danos/Devoluções',
  [StorageZone.Refrigerated]: 'Refrigerado',
  [StorageZone.Frozen]: 'Congelado',
  [StorageZone.Controlled]: 'Controlado',
  [StorageZone.Consolidation]: 'Consolidação',
  [StorageZone.Packing]: 'Packing',
};

export const ABCClassificationLabels: Record<ABCClassification, string> = {
  [ABCClassification.A]: 'A - Alto Giro',
  [ABCClassification.B]: 'B - Médio Giro',
  [ABCClassification.C]: 'C - Baixo Giro',
};
